include/picongpu/plugins/misc/ComponentNames.cpp

```diff
@@ -0,0 +1,58 @@
+/* Copyright 2020 Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "picongpu/plugins/misc/ComponentNames.hpp"
+
+#include <array>
+#include <string>
+#include <vector>
+
+
+namespace picongpu
+{
+namespace plugins
+{
+namespace misc
+{
+
+    std::vector< std::string > getComponentNames(
+        uint32_t const numComponents
+    )
+    {
+        auto result = std::vector< std::string >( numComponents );
+        for( auto i = 0u; i < result.size(); i++ )
+            result[ i ] = "component" + std::to_string( i );
+        if( result.size() <= 4 )
```

Comment:
please use `if .. elseif else` and add a comment that the case <=4 is for backwards compatibility. Please use early returns to avoid the std::copy.

Reply:
Good point, fixed and rebased.